MAINTAINERS.md

```diff
@@ -0,0 +1,144 @@
+
+### Version Branches
+When a bug is discovered in the latest release that also affects releases of the prior major version, it is necessary to backport the fix.
+
+If it is the first time that the prior major version is receiving a backported patch, then a version branch is created to track it. The version branch is named `vX.Y` where X and Y are the two most significant digits in the semantic version number. Its base is the last tag present in master for the release in question. For example, if SPIRE is on version 0.9.3, and the last 0.8 release was 0.8.4, then a `v0.8` branch is created with its base being the master commit tagged with `v0.8.4`.
+
+Once the version branch is created, the patch is either cherry picked or backported into a PR against the version branch. The version branch is maintained via the same process as the master branch, including PR approval process etc.
+
+Releases for the previous major version are made directly from its version branch. Ensure that the `CHANGELOG` is updated in both the master and the version branch to reflect the new release.
+
+### Releasing
+The SPIRE release machinery is tag-driven. When the maintainers are ready to release, a tag is pushed referencing the release commit. While the CI/CD pipeline takes care of the rest, it is important to keep an eye on its progress. If an error is encountered during this process, the release is aborted.
+
+The first two releases that a new maintainer performs must be performed under the supervision of maintainer that has already satisfied this requirement.
+
+SPIRE releases are authorized by its maintainers. When doing so, they should carefully consider the proposed release commit. Is there confidence that the changes included do not represent a compatibility concern? Have the affected codepaths been sufficiently exercised, be it by automated test suite or manual testing? Is the maintainer free of general hesitation in releasing this commit, particularly with regards to safety and security? If the answer to any of these questions is "no", then do not release.
+
+A simple majority vote is required to authorize a SPIRE release at a specific commit hash. If any maintainer feels that the result of this vote critically endangers the project or its users, they have the right to raise the matter to the SPIFFE TSC. If this occurs, the release in question MUST be frozen until the SPIFFE TSC has made a decision. To say that this would be unprecedented is an understatement. Do not take this route lightly.
+
+#### Checklist
+This section summarizes the steps necessary to execute a SPIRE release. Unless explicitly stated, the below steps must be executed in order.
+
+The following steps must be completed at least one week prior to release:
```

Comment:
```suggestion
The following steps must be completed one week prior to release:
```

Reply:
What's the motivation for this change?

Reply:
Brevity. One week firmly and briefly captures the timeline.

Reply:
Resolved offline. More succinct language without changing the lower bound, which is still one week.